bug(settings,content): UX defects with brand messaging banner

dschom opened the pull request:
## Because

- More than one banner could show up on mobile
- Tab order was wrong
- Settings header could overlap brand header

## This pull request

- Takes a different approach to injecting the banner
- Two banners are now rendered in a static matter at the top & bottom of page
- Responsive CSS is used to control the visibility of these headers
- Adds the banner to the confirm signup page for the react conversion

## Issue that this pull request solves

Closes: FXA-8352, FXA-8357, FXA-8359, FXA-8332

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

Any other information that is important to this pull request.

vpomerleau commented on packages/fxa-settings/src/components/Settings/AppLayout/index.tsx:
```diff
@@ -30,6 +31,7 @@ export const AppLayout = ({ children }: AppLayoutProps) => {
         </main>
       </div>
       <Footer />
+      <div id="body-bottom" className="block mobileLandscape:hidden" />
```
The Banner is very far down on the page for mobile, especially for 'settings' now that the banner is not sticky. I understand that's preferable for a11y - just making sure that we're okay with users potentially not seeing the banner (it will be below the initial screen view for most phones except `max` size devices).

![image](https://github.com/mozilla/fxa/assets/22231637/4c5e03d9-1dc5-41c7-a466-030762033cbe)

dschom replied:
I also raised this concern. I'll raise it again though.

dschom replied:
Thanks again for bringing this up. I think you are right it would largely go unnoticed on mobile, so for the time being I've gone ahead and fixed this by making the banner fixed in the mobile view, and thereby making it feel 'sticky'. I've tested on multiple device factors and this seems to work pretty well overall. There was a case (iPhone SE) where it obscured the screen, but I guess this is what's the close functionality is for.

LZoog replied:
I agree the sticky footer looks better than the static footer. I am kind of curious if it's better than a sticky header though - I think either way FXA-7425 would be re-opened and maybe some others, but I'm not sure what the workaround would be to fix it in all cases and would just go with whatever UX is recommending.

dschom replied:
Just to be clear the header is still static, and the footer is now sticky.

vpomerleau reviewed (changes requested):
Thanks for these fixes! I took this for a spin locally. Closing the banner is working, and screen reader + tabbing is working as intended 🎉 

I did however notice a couple of issues (commented + suggested fixes below) that would be good to fix before merging:
1) duplicated banner for some conten-server pages
2) missing banner for some pages in /settings

dschom commented on packages/fxa-settings/src/components/Settings/AppLayout/index.tsx:
```diff
@@ -20,7 +19,9 @@ export const AppLayout = ({ children }: AppLayoutProps) => {
       data-testid="app"
     >
-      {createPortal(<BrandMessaging viewName="settings" />, document.body)}
+      <div id="body-top" className="hidden mobileLandscape:block" />
```
Thank you for catching this! I was adding `showReactApp=true` and the signUpRoutes flag was true, but it wasn't rendering in react 🤦, so I was just testing the backbone version. I am still not sure why the signup routes aren't rendering in react, but I was able to validate else where. Please take another look.

LZoog reviewed (commented):
Mostly blocking r+ on the 2x banners showing up on at least React signup, secondarily because of the skip link (fine as a quick follow up if needed), but all moving in the right direction, nice job! 👍

vpomerleau commented:
A few observations from testing this morning:
- in backbone, the banner is not shown when going directly to `signup` (with email param) without going through index first - not sure if this ever happens or if important
- no banner (top or bottom) on react confirm signup code
- the banner shows up immediately on `/settings`, I was not able to replicate the delay that @LZoog noticed

dschom commented on packages/fxa-content-server/app/styles/modules/_branding.scss:
```diff
@@ -65,22 +65,50 @@
-  body.brand-messaging {
-    padding-top: inherit;
+#body-top .brand-messaging {
+  @include respond-to('small') {
```
Unfortunately @apply wouldn't work here... Not sure what's going on so I worked around it this way...

dschom commented:
Some updates were just pushed

@vpomerleau:
- The banner should now show up on the `signup` (with email param). There was a race uncovered by this page that I've put a fix in for. Nice find, I hadn't thought navigating directly would be exhibit different behavior, but it turns out it did.
- Banner has been added to the react version of ConfirmSigupCode
- Banner in footer is sticky, banner in header is not. I think raising the concern about banner being below the folding line was the right call. Thank you 👍 .

@LZoog 
- I am asking for the requirement for, _If I click on X to dismiss the banner in a particular flow (login, reg, iam) I will not see the banner again_, to be reconsidered. This would leave more flags / clutter in local storage and question if it's better.
- I fixed the banner to show up in both top and bottom. I think a last minute change caused this to regress. Essentially I lost the `className="hidden mobileLankscape:block"` bit in one of the react layout pages.
- I've moved the banner to come after the skip link.

dschom commented:
Pushed empty commit by accident. Reopening...

vpomerleau commented:
LGTM! I can confirm the issues I'd spotted are resolved :)
